### PR TITLE
fix(droid): FileOpenPicker public path

### PR DESCRIPTION
**Pull request by ajpinedam**
GitHub Issue (If applicable): closes https://github.com/unoplatform/uno/issues/16587

## PR Type

What kind of change does this PR introduce?
- Bugfix

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by ajpinedam** [comment]
Sample app running with a new Manual Test"

| Picking Videos | Picking Photos |
| :--: |  :--: |
uno_android_fileopenpicker_videos_fileinfo |  uno_android_fileopenpicker_photos_fileinfo |

File: src/Uno.UWP/Storage/Pickers/FileOpenPicker.Android.cs [hunk 1]
```diff
@@ -240,5 +250,440 @@ private string GetMimeType()
 
 		internal void RegisterOnBeforeStartActivity(Action<Intent> intentAction)
 			=> _intentAction = intentAction;
+
+		static class FileSystemUtils
+		{
+			const string FolderHash = "2203693cc04e0be7f4f024d5f9499e13";
```
Comment (jeromelaban) [inline src/Uno.UWP/Storage/Pickers/FileOpenPicker.Android.cs:256]: Where is this hash coming from?

**Comment by MartinZikmund** [comment]
I'm not sure if this is reliable, as the user can pick from any arbitrary location and the path will in many cases be not accessible via an actual path, only via that Android SAF framework. There is however a way to cache access to previously opened files (maybe it is just about storing the content URI, I don't remember exactly), so we could use that to implement FutureAccessList API

File: src/Uno.UWP/Storage/Pickers/FileOpenPicker.Android.cs (hunk 1 above)
Comment (ajpinedam) [reply src/Uno.UWP/Storage/Pickers/FileOpenPicker.Android.cs:256]: This is just a directory name. We can name it differently and have a different value, like "UnoPlatform"

**Comment by MartinZikmund** [comment]
Ah, I understand now, so this does copies the picked files into the temporary storage of the app? I'm not sure if that is the ideal, as this means that if a potentially large file is picked, it will take time to copy it as well as it will require additional space on the device. Plus, I think that this change means one cannot do actual read/write changes to picked files, as modifications will actually be done to this private copy instead, so this could even be a breaking change. I guess if that is the only way, it should be responsibility of the user to copy the file to temporary storage instead, as the path that is shown there is not the "actual" path to the source file anyway. What do you think @jeromelaban?

**Comment by MartinZikmund** [comment]
Sorry about that close, accidental :-D

**Comment by ajpinedam** [comment]
@jeromelaban  @MartinZikmund  here's the PR with the documentation changes related to this issue: https://github.com/unoplatform/uno/pull/17143

**Comment by ajpinedam** [comment]
Closing as we are not going to go this route as these changes break some other functionality. Refer to the documentation for more information.

**Pull request closed without merging**